geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java

```diff
@@ -2713,26 +2713,52 @@ public void removeAsyncEventQueueId(String asyncEventQueueId) {
     new UpdateAttributesProcessor(this).distribute();
   }
 
+  private boolean gatewaySenderIdsDifferWarningMessage;
+  private boolean asyncQueueIdsDifferWarningMessage;
+
+  /**
+   * This method used to throw an exception if the ids were different on other members.
+   * But since this check was not done until after the region had already been updated
+   * it would cause inconsistency between the contents of the region of what was sent
+   * on the gateway. So now it will simply log a warning and the caller will continue
+   * to add the event to any gateway or async event queue we have configured locally.
+   */
   @Override
   void checkSameSenderIdsAvailableOnAllNodes() {
-    List<Set<String>> senderIds =
-        getCacheDistributionAdvisor().adviseSameGatewaySenderIds(getGatewaySenderIds());
+    List senderIds =
```

Comment:
We seem to have lost the generics types here. Can we get it back? There are other places in this PR where generics are dropped.